### PR TITLE
pow implementation - now with working scalars and reverse

### PR DESCRIPTION
**Pull request by willrandship**
It's a bit ugly and big, but the tests should be much better than the last PR. the 3 conditionals are for:

scalar ^ tensor (as tensor.pow(scalar, reverse=True) )
reversed tensor^tensor (should be rare, probably should have just done return x.pow(self) in retrospect)
regular tensor^whatever (using only the tensor-first functions, so it doesn't matter if the other side is a scalar)

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/tensor.py                         7               1
------------------------------------------------------------
lines added in the tinygrad folder: 6
```

File: tinygrad/mlops.py [hunk 1]
```diff
@@ -50,6 +50,22 @@ def forward(self, x:LazyBuffer) -> LazyBuffer:
   def backward(self, grad_output:LazyBuffer) -> LazyBuffer:
     return self.ret.binary_op(BinaryOps.MUL, grad_output)
 
+class Log2(Function):
```
Comment (geohot) [inline tinygrad/mlops.py:53]: Definitely not

**Comment by willrandship** [comment]
How about this? just pow + the iterated sqrt, rsqrt.

There are also 7 removable lines in mlops.py with these changes (Pow class), but I've been leaving those.

**Comment by geohot** [comment]
Going to close, this is unreadable and not explained or tested.

**Pull request closed without merging**